engine/lite/request-response/components-utils/src/main/scala/pl/touk/nussknacker/engine/requestresponse/utils/encode/SchemaResponseEncoder.scala

```diff
@@ -0,0 +1,20 @@
+package pl.touk.nussknacker.engine.requestresponse.utils.encode
+
+import io.circe.Json
+import org.everit.json.schema.Schema
+import pl.touk.nussknacker.engine.api.validation.ValidationMode
+import pl.touk.nussknacker.engine.json.encode.BestEffortJsonSchemaEncoder
+import pl.touk.nussknacker.engine.requestresponse.api.ResponseEncoder
+
+class SchemaResponseEncoder(schema: Schema) extends ResponseEncoder[Any] {
+
+  private val bestEffortEncoder = new BestEffortJsonSchemaEncoder(ValidationMode.lax)
+
+  override def toJsonResponse(input: Any, result: List[Any]): Json = {
+    result
+      .map(bestEffortEncoder.encodeOrError(_, schema))
+      .headOption
+      .getOrElse(throw new IllegalArgumentException(s"Process did not return any result"))
```

Comment:
I don't understand this message. What process? What should return?

Reply:
its becauce interpreter return a list[any] 
      rawResult <- EitherT(invokeInterpreter(input))

Reply:
ahh, I got it. So maybe "Scenario returned empty list of results"? I was confused a bit by "process" word. I thought that it is about sth different.

Reply:
changed process to scenario - it's basically moved code so I even didn't notice there's 'process', I think its not worthy to mention there's list underneath - and I'm not sure wheather it's possible that this exception will be thrown in general...

Reply:
I think it is thrown in very simple case - when u use filter in RR scenario. After that result of micro batchprocessing is empty list which causes this exception